config/cluster_read_config.populate.yaml

```diff
@@ -12,3 +12,86 @@
 {%- endmacro %}
 
 {{ pv_stats('rhdh-postgresql', 'data-(rhdh|backstage)-(postgresql|psql)-(primary|developer-hub)-0') }}
+
+{% macro rhdh_nodejs( query ) -%}
+# Gather nodejs monitoring data about the {{ query }}
+- name: measurements.nodejs.populate.{{ query }}
+  monitoring_query: {{ query }}{ job="rhdh-metrics" }
+  monitoring_step: 15
+{%- endmacro %}
+
+{% for query in [
+'process_cpu_user_seconds_total',
+'process_cpu_system_seconds_total',
+'process_cpu_seconds_total',
+'process_resident_memory_bytes',
+'process_virtual_memory_bytes',
+'process_heap_bytes',
+'process_open_fds',
+'nodejs_eventloop_lag_seconds',
+'nodejs_eventloop_lag_mean_seconds',
+'nodejs_eventloop_lag_stddev_seconds',
+'nodejs_eventloop_lag_p90_seconds',
+'nodejs_active_resources_total',
+'nodejs_active_handles_total',
+'nodejs_active_requests_total',
+'nodejs_heap_size_total_bytes',
+'nodejs_heap_size_used_bytes',
+'nodejs_external_memory_bytes',
+'catalog_stitched_entities_count',
+'catalog_registered_locations_count',
+'catalog_relations_count',
+'catalog_processing_queue_delay_seconds_sum',
+'catalog_processing_queue_delay_seconds_count'
+] %}
+{{ rhdh_nodejs(query) }}
+{% endfor %}
+
+
+{% macro rhdh_nodejs_lst( query, label, value) -%}
+# Gather nodejs monitoring data about the {{ query }}
+- name: measurements.nodejs.populate.{{ query }}.{{ label }}.{{ 
+value }}
+  monitoring_query: {{ query }}{ {{ label }}="{{ value }}", job="rhdh-metrics" }
+  monitoring_step: 15
+{%- endmacro %}
+
+{% for query in [
```

Comment:
Here it might be easier to read just a multiple calls to that macro instead of a loop with these `if`s. And maybe learn `rhdh_nodejs_lst` to accept `value_list` instead of just `value`:
```
rhdh_nodejs_lst('catalog_processors_duration_seconds_sum', 'result', ['ok', 'failed'])
rhdh_nodejs_lst('catalog_processors_duration_seconds_count', 'result', ['ok', 'failed'])
rhdh_nodejs_lst('catalog_processing_duration_seconds_sum', 'result', ['unchanged'])
rhdh_nodejs_lst('catalog_processing_duration_seconds_count', 'result', ['unchanged'])
rhdh_nodejs_lst('nodejs_gc_duration_seconds_sum', 'kind', ['minor', 'major', 'incremental'])
rhdh_nodejs_lst('nodejs_gc_duration_seconds_count', 'kind', ['minor', 'major', 'incremental'])
rhdh_nodejs_lst('catalog_entities_count', 'kind', ['location', 'user', 'group'])
```

Reply:
Thinking about it more, are these metrics important for us? Or do we want to just collect averages?

```
catalog_processors_duration_seconds_sum{result="ok"} / catalog_processors_duration_seconds_count{result="ok"}
```

And maybe just number of failed ones?

```
rate(catalog_processors_duration_seconds_count{result="failed"}[5m])
```

I mean - instead of collecting all/lots of the individual metrics that are there, just collect these we actually care about?

Reply:
Hello @jhutar , thanks for the comments, in the latest commit I have rewritten metric collection in more readable way, as you have suggested and also added metrics for gathering rates and averages instead of just raw data.